### PR TITLE
[dev] [No PBI] - Implementing Action Buttons (Apply Filter & Clear All) on filter rails

### PR DESCRIPTION
**Pull request by fantaJinMode**
Demo Links:

Page Layout: http://localhost:8082/components/layout/page/demo

Drawer Layout: http://localhost:8082/components/surfaces/drawer (Next Step: Click on DemoDrawer, to look for filtersRail Example)

Sketch Mock Up:

image

### CODE REVIEW

File: docs/src/layout/page/demo/demoFiltersRail.jsx [hunk 1]
```diff
@@ -77,6 +77,11 @@ function DemoFiltersRail(props) {
         <Page.FiltersRail
             isOpen={isOpen}
             isScrollable
+            filterOptions={{
+                isDirty: false,
+                onClear: () => {},
+                onApply: () => {},
```
Comment (groberts314) [inline docs/src/layout/page/demo/demoFiltersRail.jsx:83]: Would it be helpful to log something to console, so consumer of demo page knows these get called?
_**Not a Blocker**_ and discretionary.

File: src/surfaces/filtersRail/filtersRail.jsx [hunk 2]
```diff
@@ -210,6 +280,32 @@ function FiltersRail(props) {
                         </ScrollBar>
                     ) :
                         children}
+                    {!isEmpty(filterOptions) && !filterOptionsRequiredKeysMissing && (
+                        <div className={actionButtonsContainerClasses}>
+                            <div className={actionButtonsClasses}>
+                                {/* eslint-disable max-len */}
+                                {/* eslint-disable jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions, jsx-a11y/anchor-is-valid */}
```
Comment (groberts314) [inline src/surfaces/filtersRail/filtersRail.jsx:287]: Do we have to suppress these?  Aren't both `<A>` and `<Button>` components compliant with all the accessibility rules?

**Review by groberts314 — APPROVED** [review]
LGTM.  Just a couple of non-blocking concerns.

File: src/surfaces/filtersRail/filtersRail.jsx (hunk 2 above)
Comment (fantaJinMode) [reply src/surfaces/filtersRail/filtersRail.jsx:287]: Yes, previously I was using HTML element `<a>` for that I was suppressing this error. With Implementation of <A>, it's not needed now. Thank you, I will remove it.

File: docs/src/layout/page/demo/demoFiltersRail.jsx (hunk 1 above)
Comment (fantaJinMode) [reply docs/src/layout/page/demo/demoFiltersRail.jsx]: Fixed. [Commit Link](https://github.com/saddlebackdev/react-cm-ui/pull/474/commits/52151f36990e262214ff2e97224c94097b3c1aec)

File: src/surfaces/filtersRail/filtersRail.jsx (hunk 2 above)
Comment (fantaJinMode) [reply src/surfaces/filtersRail/filtersRail.jsx]: Fixed. [Commit Link](https://github.com/saddlebackdev/react-cm-ui/pull/474/commits/52151f36990e262214ff2e97224c94097b3c1aec)

**Review by War777 — APPROVED** [review]
LGTM! Thanks @fantaJinMode

**Pull request merged**